XiEditor/AppDelegate.swift

```diff
@@ -140,6 +140,25 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
         } 
         return applicationDirectory
     }()
+    
+    // The default name for XiEditor's error logs
+    let defaultCoreLogName = "xi_tmp.log"
+    
+    lazy var errorLogDirectory: URL? = {
+        let logDirectory = FileManager.default.urls(
+            for: .libraryDirectory,
+            in: .userDomainMask)
+            .first?
+            .appendingPathComponent("Logs")
+            .appendingPathComponent("XiEditor")
+
+        // create XiEditor log folder on first run
+        guard logDirectory != nil else { return nil }
+        try? FileManager.default.createDirectory(at: logDirectory!,
```

Comment:
I a bit rusty with Swift but doesn't the guard above remove the need to unwrap logDirectory here?

Reply:
As mentioned above, logDirectory is nil when the /Library/ folder doesn't exist, so if it doesn't, we just return nil right there. Using guard doesn't actually unwrap the optional logDirectory value, though, so I still had to force unwrap it after the check, since I thought using `if let` would make it harder to read.

Reply:
OK, as I mentioned I haven't used Swift in a while.  Thanks for explaining.